e2e/tests/transfer/base_test.go

```diff
@@ -418,14 +412,13 @@ func (s *TransferTestSuite) TestMsgTransfer_WithMemo() {
 	chainBVersion := chainB.Config().Images[0].Version
 
 	t.Run("IBC token transfer with memo from chainA to chainB", func(t *testing.T) {
-		transferTxResp, err := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0, "memo")
-		s.Require().NoError(err)
+		transferTxResp := s.Transfer(ctx, chainA, chainAWallet, channelA.PortID, channelA.ChannelID, testvalues.DefaultTransferAmount(chainADenom), chainAAddress, chainBAddress, s.GetTimeoutHeight(ctx, chainB), 0, "memo")
 
 		if memoFeatureReleases.IsSupported(chainAVersion) {
-			s.AssertValidTxResponse(transferTxResp)
+			s.AssertTxSuccess(transferTxResp)
 		} else {
 			s.Require().Equal(uint32(2), transferTxResp.Code)
-			s.Require().Contains(transferTxResp.RawLog, "errUnknownField")
+			s.Require().Contains("errUnknownField", transferTxResp.RawLog)
```

Comment:
as far as I could tell this is coming from [`unknownproto`](https://github.com/cosmos/cosmos-sdk/blob/7b10ada768f4524edbbbcb78026903c324bdae5e/codec/unknownproto/unknown_fields.go#L313) in the sdk, couldn't find a registered error for it elsewhere.

Reply:
That's correct. I can't remember exactly where the SDK is constructing the returned error. The test is passing a sdk.Msg with an extra field the chain doesn't understand, so when it tries to unmarshal the sdk.Msg, the unknown field error is returned. I think it's fine to leave as is. It would be nice if we knew the root error though. I wonder if it is this [error](https://github.com/cosmos/cosmos-sdk/blob/7b10ada768f4524edbbbcb78026903c324bdae5e/types/errors/errors.go#L12). Ah I think I found [it](https://github.com/cosmos/cosmos-sdk/blob/7b10ada768f4524edbbbcb78026903c324bdae5e/x/auth/tx/decoder.go#L31)!

Is the updated change the right ordering? Looks like the `Contains` [godoc](https://pkg.go.dev/github.com/stretchr/testify/require#Contains) has the elem in question as the second argument?